packages/vscode-extension/src/utilities/packageManager.ts

```diff
@@ -28,6 +34,25 @@ export async function resolvePackageManager(): Promise<PackageManagerInfo | unde
   const workspacePath = findWorkspace(appRootPath);
 
   async function findPackageManager(workspace: string) {
+    const { packageManager } = getLaunchConfiguration();
+
+    if (packageManager && isPackageManager(packageManager)) {
+      return packageManager;
+    }
+
+    try {
+      const manager = require(path.join(workspace, "package.json")).packageManager;
+
+      if (manager) {
+        // e.g. yarn@3.6.4
+        const match = manager.match(/^([a-zA-Z]+)@/);
+        return match ? match[1] : "npm";
+      }
+    } catch (e) {
+      // there might be a problem while reading package.json in which case move to looking
+      // for lock files matching package managers in the workspace root
+    }
+
     const lockFiles = new Map(
       Object.entries({
         "yarn.lock": "yarn",
```

Comment:
I see that listing all of them pops up in different places. Maybe let's do some one source of truth.

Reply:
this is a different kind of mapping then in other places that is specific for that application